src/app/components/list Projects/Listsprojects.tsx

```diff
@@ -0,0 +1,74 @@
+"use client";
+import React, { useState } from "react";
+import {
+  List,
+  ListItem,
+  ListItemText,
+} from "@mui/material";
+import { AiOutlineDelete } from "react-icons/ai";
+import styles from "./style.module.scss";
+import ConfirmDelete from "../confirmDelete/ConfirmDelete";
+import { useRouter } from "next/navigation";
+import { deleteProject, getAllProjects } from "@/app/Service/ProjectServices";
+import { useMutation, useQuery, useQueryClient } from "react-query";
+
+function Listsprojects({ AllprojectsProps }: { AllprojectsProps: any }) {
+  const [openDeleteDialog, setopenDeleteDialog] = useState(false);
+  const router = useRouter();
+  const queryClient = useQueryClient();
+  const { data: allprojects } = useQuery("Allprojects", getAllProjects);
+
+  //Delete Project
+  const mutation = useMutation({
+    mutationFn: deleteProject,
+    onSuccess: () => {
+      queryClient.invalidateQueries("Allprojects");
+    },
+    onError: (error) => {
+      console.error("Failed to delete project:", error);
+    },
+  });
+  const handleDeleteProject = async (projectId: string) => {
+    setopenDeleteDialog(true);
```

Comment:
that is wrong you open dialog into async function its fatal mistake and also function that return from mutations don't need to be async await 

- also you open dialog then run function of delete before user click confirm
so what is the benefit of dialog!!
you should move the delete function to confirm function